### DISCUSSION TITLE
How to know what's the position of current argument in function calls?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1710] [Q&A] [answered]
**Question1**: If I have a function MAX(a,b), argument `a` and argument `b` are recognized to be `argument`. And MAX() is recognized as `functionCall`. How do I know what's the position of each argument in `functionCall`? 

One way is to count the number of , after left parenthesis to know the position of current parameter. But is there any other method to know the position of argument in function call?

**Question 2**: I found that if I type MAX(,), there will be grammar errors as shown below. How to delete this error msg? Can I delete it by creating a custom `ParserErrorMessageProvider`?
![image](https://github.com/user-attachments/assets/61c0c096-e026-447c-8c6d-bd8297d69f36)

https://statics.teams.cdn.office.net/evergreen-assets/safelinks/1/atp-safelinks.html

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

> If I have a function MAX(a,b), argument a and argument b are recognized to be argument. And MAX() is recognized as functionCall. How do I know what's the position of each argument in functionCall?

You can use the `$containerIndex` property of the `argument` AST node.

> I found that if I type MAX(,), there will be grammar errors as shown below. How to delete this error msg? Can I delete it by creating a custom ParserErrorMessageProvider?

I would recommend you to adjust the grammar to support adding empty arguments. Everything else is just a hack.
